src/quo/components/inputs/address_input/view.cljs

```diff
@@ -55,124 +54,134 @@
+        set-value                (rn/use-callback
+                                  (fn [new-value]
+                                    (reset! value new-value)
+                                    (on-change new-value)
+                                    (trigger-render-value new-value)))
+        on-paste                 (rn/use-callback
+                                  (fn []
+                                    (clipboard/get-string
+                                     (fn [clipboard]
+                                       (when-not (empty? clipboard)
+                                         (set-value clipboard))))))
+        on-clear                 (rn/use-callback
+                                  (fn []
+                                    (set-value "")
+                                    (set-status (if focused? :active :default))
+                                    (when on-change-text
+                                      (on-change-text ""))
+                                    (when on-clear
+                                      (on-clear)))
+                                  [focused?])
+        on-clear                 (rn/use-callback
+                                  (fn []
+                                    (set-value "")
+                                    (set-status (if focused? :active :default))
+                                    (when on-change-text
+                                      (on-change-text ""))
+                                    (when on-clear
+                                      (on-clear)))
+                                  [focused?])
+        on-scan                  (when on-scan (rn/use-callback #(on-scan set-value)))
+        on-focus                 (rn/use-callback
+                                  (fn []
+                                    (when (= (count @value) 0)
+                                      (set-status :active))
+                                    (set-focused true)
+                                    (when on-focus (on-focus))))
+        on-blur                  (rn/use-callback
+                                  (fn []
+                                    (when (= status :active)
+                                      (set-status :default))
+                                    (set-focused false)
+                                    (when on-blur (on-blur)))
+                                  [status])
+        placeholder-text-color   (rn/use-memo #(get-placeholder-text-color status theme blur?)
+                                              [status theme])]
+    [rn/view {:style (style/container container-style)}
+     [rn/text-input
+      {:accessibility-label    :address-text-input
+       :style                  (style/input-text theme)
+       :placeholder            (i18n/label :t/name-ens-or-address)
+       :placeholder-text-color placeholder-text-color
+       :value                  @value
```

Comment:
shouldn't it be `default-value` instead, since it doesn't behave like a controlled input?

Reply:
no, because it's partially controlled, 
```clojure
set-value                (rn/use-callback
                                  (fn [new-value]
                                    (reset! value new-value)
                                    (on-change new-value)
                                    (trigger-render-value new-value)))
                                    ```

trigger-render-value will trigger re-render and @value should be set to the component

Reply:
how would it behave differently if you just replaced `value` with `default-value`?

Reply:
not sure i understand what you mean, we need to set value

Reply:
i mean, i asked the team, but didn't find the answer, but default-value should be used only for default-value, it shouldn't be changed later, it's used only for default value on component initialization